app/presenters/interaction_search_results_presenter.rb

```diff
@@ -35,6 +35,34 @@ def definite_no_interactions
     results_for_view(Maybe(grouped_results[:definite_no_interactions]))
   end
 
+  def scores_for_result(result, grouped_results)
+    if @search_context == "genes"
+      scores = result.interactions.values.flatten.each_with_object({}) do |result_interaction, h|
+        overlap_count = grouped_results.map{ |r|
+          r.interactions.values.flatten.count{|other_interaction| other_interaction.drug_id == result_interaction.drug_id}
+        }.sum
+        promiscuity_count = DataModel::Interaction.where(drug_id: result_interaction.drug_id).count
```

Comment:
Are the only things differing between the two outer blocks the `drug_id` here and on 47 vs the `:gene_id` below?

If so, would it make more sense to do something like

```
identifiier  = if @search_context == 'genes'
   :drug_id
else
  :gene_id
```

and then just have the rest of the calculation be outside the conditional with `DataModel::Interaction.where(identifier => result_interation.send(identifier))`?